tools/gn

```diff
@@ -214,8 +214,69 @@ def buildtools_dir():
   return '%s-%s' % (host_os, host_cpu)
 
 
+def setup_rbe(args):
+  rbe_gn_args = {}
+  # RBE is default-off. If it is not asked for, then silently keep all default
+  # flag values.
+  if not args.rbe:
+    return rbe_gn_args
+
+  if get_host_os() not in ['linux']:
+    print(
+        'The --rbe flag has no effect. RBE is currently only supported on Linux.'
+    )
+    return rbe_gn_args
+
+  rbe_gn_args['use_rbe'] = True
+
+  # When running in CI, the recipes use their own rbe install, and take
+  # care of starting and stopping the compiler proxy.
+  running_on_luci = os.environ.get('LUCI_CONTEXT') is not None
+
+  # Bootstrap reproxy if not running in CI.
+  if not running_on_luci:
+    cipd_reclient_dir = os.path.join(
+        SRC_ROOT,
+        'buildtools',
+        buildtools_dir(),
+        'reclient',
+    )
+    bootstrap_path = os.path.join(cipd_reclient_dir, 'bootstrap')
+    reproxy_path = os.path.join(cipd_reclient_dir, 'reproxy')
+    rbe_cfg_path = os.path.join(SRC_ROOT, 'build', 'rbe.cfg')
+    bootstrap_cmd = [
+        bootstrap_path,
+        '--re_proxy=' + reproxy_path,
+        '--automatic_auth=true',
+        '--cfg=' + rbe_cfg_path,
+    ]
+    try:
+      subprocess.call(bootstrap_cmd, cwd=SRC_ROOT)
```

Comment:
From LUCI side, we expect to have a cleanup/shutdown step after task finishes. You may want to mention somewhere that users are expected to cleanup/shutdown the bootstrap later.

Reply:
There isn't a good spot to put that in the local dev workflow, and the command is too long to ask folks to run by hand, so that will need some more thought.